instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpSpanNameExtractor.java

```diff
@@ -61,19 +62,28 @@ static final class Client<REQUEST> implements SpanNameExtractor<REQUEST> {
 
     private final HttpClientAttributesGetter<REQUEST, ?> getter;
     private final Set<String> knownMethods;
+    private final Function<REQUEST, String> urlTemplateExtractor;
 
-    Client(HttpClientAttributesGetter<REQUEST, ?> getter, Set<String> knownMethods) {
+    Client(
+        HttpClientAttributesGetter<REQUEST, ?> getter,
+        Set<String> knownMethods,
+        Function<REQUEST, String> urlTemplateExtractor) {
       this.getter = getter;
       this.knownMethods = knownMethods;
+      this.urlTemplateExtractor = urlTemplateExtractor;
     }
 
     @Override
     public String extract(REQUEST request) {
       String method = getter.getHttpRequestMethod(request);
-      if (method == null || !knownMethods.contains(method)) {
+      String template = urlTemplateExtractor.apply(request);
```

Comment:
small preference to move this line down just before the last line of the method

Reply:
I did the same change for route in the server extractor.